### PR TITLE
14342 Add a way to parse rgb string values

### PR DESCRIPTION
**Pull request by jdrichards50**
Studies built in Android can now parse rgb and hex color codes.

### CODE REVIEW

File: app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java [hunk 1]
```diff
@@ -261,7 +261,26 @@ public void onClick(View v) {
...
             color.setBackgroundColor(Color.BLACK);
         } else {
-            color.setBackgroundColor(Color.parseColor(parameter.color));
+            try{
+                if(parameter.color.contains("rgb")) {
+                    String subString = parameter.color.substring(parameter.color.indexOf('(') + 1, parameter.color.indexOf(')'));
+                    String rgbColors[] = subString.split(",");
+                    int parsedColor = 0;
+                    if(rgbColors.length == 4) { // contains an alpha value
+                        parsedColor = Color.argb(Integer.parseInt(rgbColors[3].trim()), Integer.parseInt(rgbColors[0].trim()),
```
Comment (rbruckheimer) [inline app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java]: Can you simplify this a bit?  Like if there are only 3 arguments, just set the alpha to 1, and then call Color.argb regardless of whether there are 3 or 4 arguments.

Comment (jdrichards50) [reply app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java]: Sure, that's fine.

File: app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java [hunk 2]
```diff
@@ -261,7 +261,29 @@ public void onClick(View v) {
...
+                    String subString = parameter.color.substring(parameter.color.indexOf('(') + 1, parameter.color.indexOf(')'));
+                    String rgbColors[] = subString.split(",");
+                    int parsedColor = 0;
+                    int alphaValue = 255;
+
+                    // contains an alpha value
+                    if(rgbColors.length == 4) {
+                        double value = Double.parseDouble(rgbColors[3].trim());
+                        value = Math.floor(value >= 1.0 ? 255 : value * 256.0); // use 256 for floating point precision when value is less than 1.0
+                        alphaValue = (int) value;
```
Comment (rbruckheimer) [inline app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java]: I thought Math.floor was always an int.  But not important enough to change.

File: app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java [hunk 3]
```diff
@@ -261,7 +261,29 @@ public void onClick(View v) {
...
+                if(parameter.color.contains("rgb")) {
+                    String subString = parameter.color.substring(parameter.color.indexOf('(') + 1, parameter.color.indexOf(')'));
+                    String rgbColors[] = subString.split(",");
+                    int parsedColor = 0;
+                    int alphaValue = 255;
+
+                    // contains an alpha value
+                    if(rgbColors.length == 4) {
+                        double value = Double.parseDouble(rgbColors[3].trim());
+                        value = Math.floor(value >= 1.0 ? 255 : value * 256.0); // use 256 for floating point precision when value is less than 1.0
```
Comment (rbruckheimer) [inline app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java]: Was this protection necessary?  I've never seen rgba with alpha above 1.  That's like an error.

**Pull request merged**